subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# This file contains the logic for generating the decoder for the
+# 'modem_update.cddl' CDDL schema. Execute the 'cddl_gen_modem_update'
+# target to run the generation procedure.
+
+set(src_out ${NRF_DIR}/subsys/dfu/fmfu_fdev/src)
+set(include_out ${NRF_DIR}/subsys/dfu/fmfu_fdev/include)
```

Comment:
We commit this code even though its generated since it needs to be compatible with the serialization performed when building the modem firmware releases. The format is not something that will change often (it might never change), so we include the decoder files here to avoid the overhead of generating them, and to ensure that they stay compatible with the serialization in the firmware release.

The reason we add a cmake target for generating the files is to allow for re-generation if that is needed in the future.

Reply:
still, generated code should go to build folder.
Could you see if this use-case could be aligned with how openthread does things ?

OpenThread can either link to pre-built library, or build from sources.
When building from sources, an additional install target is created which can then be used to copy the library into the repository for commiting.

To aling, I roughly consider something like:
1. Build using existing sources (similar to Openthread build using prebuilt library)
2. Generate sources in built folder (similar to build Openthread from sources where library is placed in build folder)
    a. Have a dedicated install target, similar to: `install_openthread_libraries`

For more inspiration:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/openthread/CMakeLists.txt#L10
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/openthread/CMakeLists.txt#L66

Reply:
Please check it now.